src/keychain_mac.cpp

```diff
@@ -101,67 +102,80 @@ void updateError(keychain::Error &err, OSStatus status) {
     }
 }
 
-void handleCFCreateFailure(keychain::Error &err,
-                           const std::string &errorMessage) {
+void setGenericError(keychain::Error &err, const std::string &errorMessage) {
+    err = keychain::Error{};
     err.message = errorMessage;
     err.type = keychain::ErrorType::GenericError;
     err.code = -1;
 }
 
-CFStringRef createCFStringWithCString(const std::string &str,
-                                      keychain::Error &err) {
-    CFStringRef result = CFStringCreateWithCString(
-        kCFAllocatorDefault, str.c_str(), kCFStringEncodingUTF8);
-    if (result == NULL) {
-        handleCFCreateFailure(err, "Failed to create CFString");
+/*! \brief Helper to manage the lifetime of CFObjects */
+template <typename T> struct ScopedCFRef {
+  public:
+    explicit ScopedCFRef(T &&ref) : _ref(std::move(ref)) {}
+    ~ScopedCFRef() {
+        if (_ref)
+            CFRelease(_ref);
     }
+
+    ScopedCFRef(ScopedCFRef &&) noexcept = default;
+    ScopedCFRef(const ScopedCFRef &) = delete;
+    ScopedCFRef &operator=(const ScopedCFRef &) = delete;
+    ScopedCFRef &operator=(ScopedCFRef &&) = delete;
```

Comment:
```suggestion
    ScopedCFRef &operator=(ScopedCFRef &&other) {
        if(this != &other) {
            maybe_release();
            _ref = other._ref;
            other._ref = nullptr;
        }
        return *this;
    }
```

... move-assignment is nice, why would you drop it? Just make sure that you destroy the object reference you're  overwriting with the move assignment. 😏